lib/GenomeAnnotationAPI/GenomeAnnotationUtil.py

```diff
@@ -0,0 +1,311 @@
+import os
+import hashlib
+from collections import defaultdict
+from repoze.lru import lru_cache
+
+from Workspace.WorkspaceClient import Workspace
+from DataFileUtil.DataFileUtilClient import DataFileUtil
+
+
+class GenomeIAnnotationUtil:
+    def __init__(self, services):
+        self.ws = Workspace(services['workspace_service_url'])
+        self.handle_url = services['handle_service_url']
+        self.shock_url = services['shock_service_url']
+        self.sw_url = services['service_wizard_url']
+        self.dfu = DataFileUtil(os.environ['SDK_CALLBACK_URL'])
+        self.taxon_wsname = 'ReferenceTaxons'
+        self._valid_filters = ["type_list", "region_list", "function_list",
+                               "alias_list"]
+        self._valid_groups = ["type", "region", "function", "alias"]
+
+    @lru_cache(maxsize=8)
+    def get_genome(self, genome_ref):
+        return self.dfu.get_objects(
+            {'object_refs': [genome_ref]}
+        )['data'][0]['data']
+
+    def get_taxon_ref(self, genome_ref):
+        return self.ws.get_objects2({'objects': [
+            {'ref': genome_ref, 'included': ['/taxon_ref']}
+        ]})['data'][0]['data']['taxon_ref']
+
+    def get_assembly_ref(self, genome_ref):
```

Comment:
should it include contigset_ref like the original function?

Reply:
done